urllib3/request.py

```diff
@@ -40,6 +41,12 @@ class RequestMethods(object):
 
     def __init__(self, headers=None):
         self.headers = headers or {}
+        try: 
+            version = sys.modules['urllib3'].__version__
+            self.headers.update({'User-Agent': 'python-urllib3/%s' % version})
```

Comment:
We would want to do a `setdefault()` here instead of `update()` so that if a user sets a `User-Agent` header ours doesn't take precedence. We should have a test case for this as well.

Reply:
We actually don't want to be here at all; the object-default headers don't get used if the user passes a set of headers in. A good place to set it might be in `extra_kw` a bit lower down.

Reply:
Actually, check out `make_headers` in `util/request.py` as a potential very good spot for this.